tests/python/contrib/test_hexagon/test_meta_schedule.py

```diff
@@ -388,24 +387,29 @@ def test_conv2d_relay_auto_schedule(hexagon_launcher):
 
     ref = rt_mod_ref.get_output(0).numpy()
 
-    config = ms.TuneConfig(
-        strategy="replay_trace",
-        num_trials_per_iter=8,
-        max_trials_per_task=8,
-        max_trials_global=8,
-    )
-
     with tempfile.TemporaryDirectory() as work_dir:
-        executor = Executor("graph", {"link-params": True})
-        lib = ms.tune_relay(
+        database = ms.relay_integration.tune_relay(
```

Comment:
@masahi would be great to have your eyes on the hexagon-related changes in case anything goes wrong :-)

Reply:
This is a bit tricky. Both `tune_relay` and `compile_relay` need to know if `link-params` is enabled or not. Using `relay.FuseOps.link_params` was a temp workaround, since I didn't want to change the `extract_task` API. But if we are changing it, I suggest adding `executor` argument to all functions that directly or indirectly call `FuseOps` (see `FuseOps.cc` and how `link-params` attribute is queried). Previously they were task extraction and `relay.build`, but now we also want to add `exeuctor` to `tune_relay` as well. 

Also see my change in https://github.com/apache/tvm/pull/12854. I need `executor` change in `tune.py` to be restored.

Reply:
Thank you for your detailed explanation Masa! I looked into your PR and it looks very informative - added `executor` to `compile_relay` API. However, I'm not confident if I'm doing things exactly right, so it would be super helpful if you could take a second look for proofreading!!